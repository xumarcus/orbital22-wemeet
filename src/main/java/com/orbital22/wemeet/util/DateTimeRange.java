package com.orbital22.wemeet.util;

import lombok.NonNull;

import java.time.LocalDateTime;

public interface DateTimeRange {
  @NonNull
  LocalDateTime getStartDateTime();

  @NonNull
  LocalDateTime getEndDateTime();

  default boolean contains(DateTimeRange other) {
    return getStartDateTime().isBefore(other.getStartDateTime())
        && getEndDateTime().isAfter(other.getEndDateTime());
  }
}
