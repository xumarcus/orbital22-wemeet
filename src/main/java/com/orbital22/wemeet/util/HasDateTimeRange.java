package com.orbital22.wemeet.util;

import lombok.NonNull;

import java.time.LocalDateTime;

public interface HasDateTimeRange {
  @NonNull
  LocalDateTime getStartDateTime();

  @NonNull
  LocalDateTime getEndDateTime();

  static boolean isValid(HasDateTimeRange dateTimeRange) {
    return dateTimeRange.getStartDateTime().isBefore(dateTimeRange.getEndDateTime());
  }

  default boolean containsRange(HasDateTimeRange other) {
    return getStartDateTime().isBefore(other.getStartDateTime())
        && getEndDateTime().isAfter(other.getEndDateTime());
  }
}
