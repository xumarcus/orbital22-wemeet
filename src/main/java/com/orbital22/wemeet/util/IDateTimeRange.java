package com.orbital22.wemeet.util;

import lombok.NonNull;
import org.apache.commons.lang3.Range;

import java.time.LocalDateTime;
import java.time.chrono.ChronoLocalDateTime;

import static org.apache.commons.lang3.Range.between;

public interface IDateTimeRange {
  @NonNull
  LocalDateTime getStartDateTime();

  @NonNull
  LocalDateTime getEndDateTime();

  default boolean isValid() {
    return getStartDateTime().isBefore(getEndDateTime());
  }

  default Range<ChronoLocalDateTime<?>> getChronoLocalDateTimeRange() {
    return between(getStartDateTime(), getEndDateTime());
  }
}
