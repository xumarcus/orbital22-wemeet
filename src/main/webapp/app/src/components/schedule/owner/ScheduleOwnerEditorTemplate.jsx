import * as React from 'react'
import _ from 'lodash'
import FieldEditorRow from '../editor/FieldEditorRow'
import DateTimePickerEditorRow from '../editor/DateTimePickerEditorRow'
import EditorListWithAvatar from '../editor/EditorListWithAvatar'

const ScheduleOwnerEditorTemplate = (props) => {
  if (_.isEmpty(props)) {
    return <div />  // Throw?
  }

  const {
    id: timeSlotId,
    // Id: syncfusionId,
    startDateTime,
    endDateTime,
    capacity,
    timeSlotUserInfos
  } = props

  return (
    <table
      className='custom-event-editor'
      style={{ width: '100%', cellpadding: '5' }}
    >
      <tbody>
        <FieldEditorRow
          name='timeSlotId'
          type='number' value={timeSlotId} disabled hidden
        />
        <FieldEditorRow
          label='Capacity' name='capacity'
          defaultValue={capacity} type='number'
        />
        <DateTimePickerEditorRow
          label='From' name='startDateTime' value={startDateTime}
        />
        <DateTimePickerEditorRow
          label='To' name='endDateTime' value={endDateTime}
        />
        <EditorListWithAvatar
          label='Picked'
          items={timeSlotUserInfos.filter(({ available }) => available)
            .map(info => info.user.email)}
        />
        <EditorListWithAvatar
          label='Allocated'
          items={timeSlotUserInfos.filter(({ picked }) => picked /* FIXME fieldName */)
            .map(info => info.user.email)}
        />
      </tbody>
    </table>
  )
}

export default ScheduleOwnerEditorTemplate
