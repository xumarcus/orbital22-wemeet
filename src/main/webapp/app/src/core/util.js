// In development, resp.type === 'cors'
// In production, resp.type is per normal
// TODO handle fullyAuthenticated

const ajax = (method, data) => async (uri) => {
    // Spring POST /login accepts formData by default.
    // POST /logout does not take any parameters.
    const isFormData = uri.endsWith("login");
    const resp = await fetch(uri, {
        body: isFormData ? data : JSON.stringify(data),
        headers: {
            "X-XSRF-TOKEN": csrfToken(),
            ...(isFormData ? undefined : {"Content-Type": "application/json"})
        },
        method,
        redirect: "follow",
    });
    if (resp.ok) {
        const location = resp.headers.get('Location');
        if (location != null) {
            // Fetch created resource
            return ajax('GET')(location);
        }
        return resp.json();
    } else {
        throw new Error(resp.statusText, { cause: await resp.json() });
    }
};

const csrfToken = () => document.cookie.replace(/(?:^|.*;\s*)XSRF-TOKEN\s*=\s*([^;]*).*$|^.*$/, '$1');

export default ajax;