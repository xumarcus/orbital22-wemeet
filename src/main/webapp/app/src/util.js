const ajax = (method) => (uri) => {
    const headers = {'Content-Type': 'application/json'};
    return fetch(uri, { method, headers })
        .then(resp => resp.json());
}
export default ajax;