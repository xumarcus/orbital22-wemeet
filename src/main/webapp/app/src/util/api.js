import { useEffect } from 'react';

export function ajax(method, uri) {
    return fetch(uri, { method, headers: {'Content-Type': 'application/json'} });
}

export function useAjax(method, uri, setState, callback) {
    useEffect(() => {
        const fetchData = async () => {
            const response = await ajax(method, uri);
            if (response.ok) {
                const json = await response.json();
                setState(json);
            } else {
                callback();
            }
        }
        fetchData().then();
    }, []);
}