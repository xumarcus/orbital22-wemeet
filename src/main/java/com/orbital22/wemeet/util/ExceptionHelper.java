package com.orbital22.wemeet.util;

import org.jetbrains.annotations.Nullable;
import org.springframework.data.rest.core.RepositoryConstraintViolationException;
import org.springframework.validation.Errors;
import org.springframework.validation.MapBindingResult;

import java.util.HashMap;

public interface ExceptionHelper {
  static RepositoryConstraintViolationException from(
          Object object, @Nullable String field, String errorCode) {
    return from(object, field, errorCode, null);
  }

  static RepositoryConstraintViolationException from(
          Object object, @Nullable String field, String errorCode, @Nullable String defaultMessage) {
    Errors errors = new MapBindingResult(new HashMap<>(), object.getClass().getName());
    Object[] errorArgs = new Object[] {object};

    if (field != null) {
      errors.rejectValue(field, errorCode, errorArgs, defaultMessage);
    } else {
      errors.reject(errorCode, errorArgs, defaultMessage);
    }

    return new RepositoryConstraintViolationException(errors);
  }
}
