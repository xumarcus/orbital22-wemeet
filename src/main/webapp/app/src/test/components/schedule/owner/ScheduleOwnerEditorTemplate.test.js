import React from 'react'
import { render } from '@testing-library/react'
import { SAMPLE_EVENT } from '../../../mocks/events'
import { setupForSyncfusionTest } from '../../../test-core/util'
import ScheduleOwnerEditorTemplate
  from '../../../../components/schedule/owner/ScheduleOwnerEditorTemplate'
import { beforeAll, describe, expect, it } from '@jest/globals'

describe('ScheduleOwnerEditorTemplate', () => {
  beforeAll(setupForSyncfusionTest)

  it('should render correctly without data', () => {
    const { asFragment } = render(<ScheduleOwnerEditorTemplate
      id={1}
      timeSlotUserInfos={[]}
      {...SAMPLE_EVENT}
                                  />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('should render correctly with user picks', () => {
    const { asFragment, getByText } = render(<ScheduleOwnerEditorTemplate
      id={1}
      timeSlotUserInfos={[{ available: true, user: { email: 'test@test.com' } }]}
      {...SAMPLE_EVENT}
                                             />)
    expect(getByText('test@test.com')).toBeInTheDocument()
    expect(asFragment()).toMatchSnapshot()
  })

  it('should render correctly with solver allocations', () => {
    const { asFragment, getByText } = render(<ScheduleOwnerEditorTemplate
      id={1}
      timeSlotUserInfos={[
        {
          picked: true,
          user: { email: 'test@test.com' }
        }]}
      {...SAMPLE_EVENT}
                                             />)
    expect(getByText('test@test.com')).toBeInTheDocument()
    expect(asFragment()).toMatchSnapshot()
  })
})
