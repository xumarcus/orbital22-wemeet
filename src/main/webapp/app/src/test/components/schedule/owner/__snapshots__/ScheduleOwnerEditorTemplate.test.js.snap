// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScheduleOwnerEditorTemplate should render correctly with allocations 1`] = `
<DocumentFragment>
  <table
    class="custom-event-editor"
    style="width: 100%;"
  >
    <tbody>
      <tr>
        <td
          class="e-textlabel"
        />
        <td
          colspan="4"
        >
          <input
            class="e-field e-input"
            disabled=""
            hidden=""
            id="timeSlotId"
            name="timeSlotId"
            style="width: 100%;"
            type="number"
            value="1"
          />
        </td>
      </tr>
      <tr>
        <td
          class="e-textlabel"
        >
          Capacity
        </td>
        <td
          colspan="4"
        >
          <input
            class="e-field e-input"
            disabled=""
            id="capacity"
            name="capacity"
            style="width: 100%;"
            value="1337"
          />
        </td>
      </tr>
      <tr>
        <td
          class="e-textlabel"
        >
          Available
        </td>
        <td
          colspan="4"
        >
          <div
            aria-checked="false"
            class="e-switch-wrapper e-wrapper"
          >
            <input
              class="e-field e-control e-switch e-lib"
              data-name="available"
              id="available"
              type="checkbox"
            />
            <span
              class="e-switch-inner"
            >
              <span
                class="e-switch-on"
              />
              <span
                class="e-switch-off"
              />
            </span>
            <span
              class="e-switch-handle"
            />
          </div>
        </td>
      </tr>
      <tr>
        <td
          class="e-textlabel"
        >
          Rank
        </td>
        <td
          colspan="4"
        >
          <input
            class="e-field e-input"
            disabled=""
            id="rank"
            name="rank"
            style="width: 100%;"
            type="number"
            value=""
          />
        </td>
      </tr>
      <tr>
        <td
          class="e-textlabel"
        >
          From
        </td>
        <td
          colspan="4"
        >
          <span
            class="e-input-group e-control-wrapper e-disabled e-valid-input e-datetime-wrapper"
            style="width: 100%;"
          >
            <input
              aria-activedescendant="null"
              aria-atomic="true"
              aria-disabled="true"
              aria-expanded="false"
              aria-haspopup="true"
              aria-invalid="false"
              aria-live="assertive"
              aria-owns="startDateTime_options"
              autocapitalize="off"
              autocomplete="off"
              autocorrect="off"
              class="e-field e-control e-datetimepicker e-lib e-input e-disabled e-keyboard"
              data-name="startDateTime"
              disabled="disabled"
              id="startDateTime"
              name="startDateTime"
              role="combobox"
              spellcheck="false"
              tabindex="-1"
              value="09/06/19 02:20 PM"
            />
            <span
              aria-label="close"
              class="e-clear-icon e-clear-icon-hide"
              role="button"
            />
            <span
              aria-label="select"
              class="e-input-group-icon e-date-icon e-icons"
            />
            <span
              class="e-input-group-icon e-time-icon e-icons"
            />
          </span>
        </td>
      </tr>
      <tr>
        <td
          class="e-textlabel"
        >
          To
        </td>
        <td
          colspan="4"
        >
          <span
            class="e-input-group e-control-wrapper e-disabled e-valid-input e-datetime-wrapper"
            style="width: 100%;"
          >
            <input
              aria-activedescendant="null"
              aria-atomic="true"
              aria-disabled="true"
              aria-expanded="false"
              aria-haspopup="true"
              aria-invalid="false"
              aria-live="assertive"
              aria-owns="endDateTime_options"
              autocapitalize="off"
              autocomplete="off"
              autocorrect="off"
              class="e-field e-control e-datetimepicker e-lib e-input e-disabled e-keyboard"
              data-name="endDateTime"
              disabled="disabled"
              id="endDateTime"
              name="endDateTime"
              role="combobox"
              spellcheck="false"
              tabindex="-1"
              value="09/06/19 10:00 PM"
            />
            <span
              aria-label="close"
              class="e-clear-icon e-clear-icon-hide"
              role="button"
            />
            <span
              aria-label="select"
              class="e-input-group-icon e-date-icon e-icons"
            />
            <span
              class="e-input-group-icon e-time-icon e-icons"
            />
          </span>
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;

exports[`ScheduleOwnerEditorTemplate should render correctly with picks 1`] = `
<DocumentFragment>
  <table
    class="custom-event-editor"
    style="width: 100%;"
  >
    <tbody>
      <tr>
        <td
          class="e-textlabel"
        />
        <td
          colspan="4"
        >
          <input
            class="e-field e-input"
            disabled=""
            hidden=""
            id="timeSlotId"
            name="timeSlotId"
            style="width: 100%;"
            type="number"
            value="1"
          />
        </td>
      </tr>
      <tr>
        <td
          class="e-textlabel"
        >
          Capacity
        </td>
        <td
          colspan="4"
        >
          <input
            class="e-field e-input"
            id="capacity"
            name="capacity"
            style="width: 100%;"
            type="number"
            value="1337"
          />
        </td>
      </tr>
      <tr>
        <td
          class="e-textlabel"
        >
          From
        </td>
        <td
          colspan="4"
        >
          <span
            class="e-input-group e-control-wrapper e-valid-input e-datetime-wrapper"
            style="width: 100%;"
          >
            <input
              aria-activedescendant="null"
              aria-atomic="true"
              aria-disabled="false"
              aria-expanded="false"
              aria-haspopup="true"
              aria-invalid="false"
              aria-live="assertive"
              aria-owns="startDateTime_options"
              autocapitalize="off"
              autocomplete="off"
              autocorrect="off"
              class="e-field e-control e-datetimepicker e-lib e-input e-keyboard"
              data-name="startDateTime"
              id="startDateTime"
              name="startDateTime"
              role="combobox"
              spellcheck="false"
              tabindex="0"
              value="09/06/19 02:20 PM"
            />
            <span
              aria-label="close"
              class="e-clear-icon e-clear-icon-hide"
              role="button"
            />
            <span
              aria-label="select"
              class="e-input-group-icon e-date-icon e-icons"
            />
            <span
              class="e-input-group-icon e-time-icon e-icons"
            />
          </span>
        </td>
      </tr>
      <tr>
        <td
          class="e-textlabel"
        >
          To
        </td>
        <td
          colspan="4"
        >
          <span
            class="e-input-group e-control-wrapper e-valid-input e-datetime-wrapper"
            style="width: 100%;"
          >
            <input
              aria-activedescendant="null"
              aria-atomic="true"
              aria-disabled="false"
              aria-expanded="false"
              aria-haspopup="true"
              aria-invalid="false"
              aria-live="assertive"
              aria-owns="endDateTime_options"
              autocapitalize="off"
              autocomplete="off"
              autocorrect="off"
              class="e-field e-control e-datetimepicker e-lib e-input e-keyboard"
              data-name="endDateTime"
              id="endDateTime"
              name="endDateTime"
              role="combobox"
              spellcheck="false"
              tabindex="0"
              value="09/06/19 10:00 PM"
            />
            <span
              aria-label="close"
              class="e-clear-icon e-clear-icon-hide"
              role="button"
            />
            <span
              aria-label="select"
              class="e-input-group-icon e-date-icon e-icons"
            />
            <span
              class="e-input-group-icon e-time-icon e-icons"
            />
          </span>
        </td>
      </tr>
      <tr>
        <td
          class="e-textlabel"
          style="padding-top: 16px; padding-bottom: 16px;"
        >
          Picked
        </td>
        <td
          colspan="4"
        >
          <span
            class="MuiTypography-root MuiTypography-body css-37jdci-MuiTypography-root"
          >
            No records to display
          </span>
        </td>
      </tr>
      <tr>
        <td
          class="e-textlabel"
          style="padding-top: 16px; padding-bottom: 16px;"
        >
          Allocated
        </td>
        <td
          colspan="4"
        >
          <ul
            class="MuiList-root MuiList-padding MuiList-dense css-h4y409-MuiList-root"
          >
            <li
              class="MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-padding css-6lnq02-MuiListItem-root"
            >
              <div
                class="MuiListItemAvatar-root css-1e9lk82-MuiListItemAvatar-root"
              >
                <div
                  class="MuiAvatar-root MuiAvatar-circular css-1azxqog-MuiAvatar-root"
                >
                  <img
                    class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
                    src="/avatar-does-not-exist"
                  />
                </div>
              </div>
              <div
                class="MuiListItemText-root MuiListItemText-dense css-tlelie-MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiTypography-body2 MuiListItemText-primary css-et1ao3-MuiTypography-root"
                >
                  test@test.com
                </span>
              </div>
            </li>
          </ul>
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;

exports[`ScheduleOwnerEditorTemplate should render correctly with solver allocations 1`] = `
<DocumentFragment>
  <table
    class="custom-event-editor"
    style="width: 100%;"
  >
    <tbody>
      <tr>
        <td
          class="e-textlabel"
        />
        <td
          colspan="4"
        >
          <input
            class="e-field e-input"
            disabled=""
            hidden=""
            id="timeSlotId"
            name="timeSlotId"
            style="width: 100%;"
            type="number"
            value="1"
          />
        </td>
      </tr>
      <tr>
        <td
          class="e-textlabel"
        >
          Capacity
        </td>
        <td
          colspan="4"
        >
          <input
            class="e-field e-input"
            id="capacity"
            name="capacity"
            style="width: 100%;"
            type="number"
            value="1337"
          />
        </td>
      </tr>
      <tr>
        <td
          class="e-textlabel"
        >
          From
        </td>
        <td
          colspan="4"
        >
          <span
            class="e-input-group e-control-wrapper e-valid-input e-datetime-wrapper"
            style="width: 100%;"
          >
            <input
              aria-activedescendant="null"
              aria-atomic="true"
              aria-disabled="false"
              aria-expanded="false"
              aria-haspopup="true"
              aria-invalid="false"
              aria-live="assertive"
              aria-owns="startDateTime_options"
              autocapitalize="off"
              autocomplete="off"
              autocorrect="off"
              class="e-field e-control e-datetimepicker e-lib e-input e-keyboard"
              data-name="startDateTime"
              id="startDateTime"
              name="startDateTime"
              role="combobox"
              spellcheck="false"
              tabindex="0"
              value="09/06/19 02:20 PM"
            />
            <span
              aria-label="close"
              class="e-clear-icon e-clear-icon-hide"
              role="button"
            />
            <span
              aria-label="select"
              class="e-input-group-icon e-date-icon e-icons"
            />
            <span
              class="e-input-group-icon e-time-icon e-icons"
            />
          </span>
        </td>
      </tr>
      <tr>
        <td
          class="e-textlabel"
        >
          To
        </td>
        <td
          colspan="4"
        >
          <span
            class="e-input-group e-control-wrapper e-valid-input e-datetime-wrapper"
            style="width: 100%;"
          >
            <input
              aria-activedescendant="null"
              aria-atomic="true"
              aria-disabled="false"
              aria-expanded="false"
              aria-haspopup="true"
              aria-invalid="false"
              aria-live="assertive"
              aria-owns="endDateTime_options"
              autocapitalize="off"
              autocomplete="off"
              autocorrect="off"
              class="e-field e-control e-datetimepicker e-lib e-input e-keyboard"
              data-name="endDateTime"
              id="endDateTime"
              name="endDateTime"
              role="combobox"
              spellcheck="false"
              tabindex="0"
              value="09/06/19 10:00 PM"
            />
            <span
              aria-label="close"
              class="e-clear-icon e-clear-icon-hide"
              role="button"
            />
            <span
              aria-label="select"
              class="e-input-group-icon e-date-icon e-icons"
            />
            <span
              class="e-input-group-icon e-time-icon e-icons"
            />
          </span>
        </td>
      </tr>
      <tr>
        <td
          class="e-textlabel"
          style="padding-top: 16px; padding-bottom: 16px;"
        >
          Picked
        </td>
        <td
          colspan="4"
        >
          <span
            class="MuiTypography-root MuiTypography-body css-37jdci-MuiTypography-root"
          >
            No records to display
          </span>
        </td>
      </tr>
      <tr>
        <td
          class="e-textlabel"
          style="padding-top: 16px; padding-bottom: 16px;"
        >
          Allocated
        </td>
        <td
          colspan="4"
        >
          <ul
            class="MuiList-root MuiList-padding MuiList-dense css-h4y409-MuiList-root"
          >
            <li
              class="MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-padding css-6lnq02-MuiListItem-root"
            >
              <div
                class="MuiListItemAvatar-root css-1e9lk82-MuiListItemAvatar-root"
              >
                <div
                  class="MuiAvatar-root MuiAvatar-circular css-1azxqog-MuiAvatar-root"
                >
                  <img
                    class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
                    src="/avatar-does-not-exist"
                  />
                </div>
              </div>
              <div
                class="MuiListItemText-root MuiListItemText-dense css-tlelie-MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiTypography-body2 MuiListItemText-primary css-et1ao3-MuiTypography-root"
                >
                  test@test.com
                </span>
              </div>
            </li>
          </ul>
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;

exports[`ScheduleOwnerEditorTemplate should render correctly with user picks 1`] = `
<DocumentFragment>
  <table
    class="custom-event-editor"
    style="width: 100%;"
  >
    <tbody>
      <tr>
        <td
          class="e-textlabel"
        />
        <td
          colspan="4"
        >
          <input
            class="e-field e-input"
            disabled=""
            hidden=""
            id="timeSlotId"
            name="timeSlotId"
            style="width: 100%;"
            type="number"
            value="1"
          />
        </td>
      </tr>
      <tr>
        <td
          class="e-textlabel"
        >
          Capacity
        </td>
        <td
          colspan="4"
        >
          <input
            class="e-field e-input"
            id="capacity"
            name="capacity"
            style="width: 100%;"
            type="number"
            value="1337"
          />
        </td>
      </tr>
      <tr>
        <td
          class="e-textlabel"
        >
          From
        </td>
        <td
          colspan="4"
        >
          <span
            class="e-input-group e-control-wrapper e-valid-input e-datetime-wrapper"
            style="width: 100%;"
          >
            <input
              aria-activedescendant="null"
              aria-atomic="true"
              aria-disabled="false"
              aria-expanded="false"
              aria-haspopup="true"
              aria-invalid="false"
              aria-live="assertive"
              aria-owns="startDateTime_options"
              autocapitalize="off"
              autocomplete="off"
              autocorrect="off"
              class="e-field e-control e-datetimepicker e-lib e-input e-keyboard"
              data-name="startDateTime"
              id="startDateTime"
              name="startDateTime"
              role="combobox"
              spellcheck="false"
              tabindex="0"
              value="09/06/19 02:20 PM"
            />
            <span
              aria-label="close"
              class="e-clear-icon e-clear-icon-hide"
              role="button"
            />
            <span
              aria-label="select"
              class="e-input-group-icon e-date-icon e-icons"
            />
            <span
              class="e-input-group-icon e-time-icon e-icons"
            />
          </span>
        </td>
      </tr>
      <tr>
        <td
          class="e-textlabel"
        >
          To
        </td>
        <td
          colspan="4"
        >
          <span
            class="e-input-group e-control-wrapper e-valid-input e-datetime-wrapper"
            style="width: 100%;"
          >
            <input
              aria-activedescendant="null"
              aria-atomic="true"
              aria-disabled="false"
              aria-expanded="false"
              aria-haspopup="true"
              aria-invalid="false"
              aria-live="assertive"
              aria-owns="endDateTime_options"
              autocapitalize="off"
              autocomplete="off"
              autocorrect="off"
              class="e-field e-control e-datetimepicker e-lib e-input e-keyboard"
              data-name="endDateTime"
              id="endDateTime"
              name="endDateTime"
              role="combobox"
              spellcheck="false"
              tabindex="0"
              value="09/06/19 10:00 PM"
            />
            <span
              aria-label="close"
              class="e-clear-icon e-clear-icon-hide"
              role="button"
            />
            <span
              aria-label="select"
              class="e-input-group-icon e-date-icon e-icons"
            />
            <span
              class="e-input-group-icon e-time-icon e-icons"
            />
          </span>
        </td>
      </tr>
      <tr>
        <td
          class="e-textlabel"
          style="padding-top: 16px; padding-bottom: 16px;"
        >
          Picked
        </td>
        <td
          colspan="4"
        >
          <ul
            class="MuiList-root MuiList-padding MuiList-dense css-h4y409-MuiList-root"
          >
            <li
              class="MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-padding css-6lnq02-MuiListItem-root"
            >
              <div
                class="MuiListItemAvatar-root css-1e9lk82-MuiListItemAvatar-root"
              >
                <div
                  class="MuiAvatar-root MuiAvatar-circular css-1azxqog-MuiAvatar-root"
                >
                  <img
                    class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
                    src="/avatar-does-not-exist"
                  />
                </div>
              </div>
              <div
                class="MuiListItemText-root MuiListItemText-dense css-tlelie-MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiTypography-body2 MuiListItemText-primary css-et1ao3-MuiTypography-root"
                >
                  test@test.com
                </span>
              </div>
            </li>
          </ul>
        </td>
      </tr>
      <tr>
        <td
          class="e-textlabel"
          style="padding-top: 16px; padding-bottom: 16px;"
        >
          Allocated
        </td>
        <td
          colspan="4"
        >
          <span
            class="MuiTypography-root MuiTypography-body css-37jdci-MuiTypography-root"
          >
            No records to display
          </span>
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;

exports[`ScheduleOwnerEditorTemplate should render correctly without data 1`] = `
<DocumentFragment>
  <table
    class="custom-event-editor"
    style="width: 100%;"
  >
    <tbody>
      <tr>
        <td
          class="e-textlabel"
        />
        <td
          colspan="4"
        >
          <input
            class="e-field e-input"
            disabled=""
            hidden=""
            id="timeSlotId"
            name="timeSlotId"
            style="width: 100%;"
            type="number"
            value="1"
          />
        </td>
      </tr>
      <tr>
        <td
          class="e-textlabel"
        >
          Capacity
        </td>
        <td
          colspan="4"
        >
          <input
            class="e-field e-input"
            id="capacity"
            name="capacity"
            style="width: 100%;"
            type="number"
            value="1337"
          />
        </td>
      </tr>
      <tr>
        <td
          class="e-textlabel"
        >
          From
        </td>
        <td
          colspan="4"
        >
          <span
            class="e-input-group e-control-wrapper e-valid-input e-datetime-wrapper"
            style="width: 100%;"
          >
            <input
              aria-activedescendant="null"
              aria-atomic="true"
              aria-disabled="false"
              aria-expanded="false"
              aria-haspopup="true"
              aria-invalid="false"
              aria-live="assertive"
              aria-owns="startDateTime_options"
              autocapitalize="off"
              autocomplete="off"
              autocorrect="off"
              class="e-field e-control e-datetimepicker e-lib e-input e-keyboard"
              data-name="startDateTime"
              id="startDateTime"
              name="startDateTime"
              role="combobox"
              spellcheck="false"
              tabindex="0"
              value="09/06/19 02:20 PM"
            />
            <span
              aria-label="close"
              class="e-clear-icon e-clear-icon-hide"
              role="button"
            />
            <span
              aria-label="select"
              class="e-input-group-icon e-date-icon e-icons"
            />
            <span
              class="e-input-group-icon e-time-icon e-icons"
            />
          </span>
        </td>
      </tr>
      <tr>
        <td
          class="e-textlabel"
        >
          To
        </td>
        <td
          colspan="4"
        >
          <span
            class="e-input-group e-control-wrapper e-valid-input e-datetime-wrapper"
            style="width: 100%;"
          >
            <input
              aria-activedescendant="null"
              aria-atomic="true"
              aria-disabled="false"
              aria-expanded="false"
              aria-haspopup="true"
              aria-invalid="false"
              aria-live="assertive"
              aria-owns="endDateTime_options"
              autocapitalize="off"
              autocomplete="off"
              autocorrect="off"
              class="e-field e-control e-datetimepicker e-lib e-input e-keyboard"
              data-name="endDateTime"
              id="endDateTime"
              name="endDateTime"
              role="combobox"
              spellcheck="false"
              tabindex="0"
              value="09/06/19 10:00 PM"
            />
            <span
              aria-label="close"
              class="e-clear-icon e-clear-icon-hide"
              role="button"
            />
            <span
              aria-label="select"
              class="e-input-group-icon e-date-icon e-icons"
            />
            <span
              class="e-input-group-icon e-time-icon e-icons"
            />
          </span>
        </td>
      </tr>
      <tr>
        <td
          class="e-textlabel"
          style="padding-top: 16px; padding-bottom: 16px;"
        >
          Picked
        </td>
        <td
          colspan="4"
        >
          <span
            class="MuiTypography-root MuiTypography-body css-37jdci-MuiTypography-root"
          >
            No records to display
          </span>
        </td>
      </tr>
      <tr>
        <td
          class="e-textlabel"
          style="padding-top: 16px; padding-bottom: 16px;"
        >
          Allocated
        </td>
        <td
          colspan="4"
        >
          <span
            class="MuiTypography-root MuiTypography-body css-37jdci-MuiTypography-root"
          >
            No records to display
          </span>
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
